lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1035,11 +1005,11 @@ def customize_vm(self, vm_obj):
                 ident.userData.orgName = str(self.params['customizations']['productid'])
 
             if 'joindomain' in self.params['customizations']:
-                ident.identification.domainadmin = str(self.params['customizations']['domainadmin'])
-                ident.identification.domainadminpassword = str(self.params['customizations']['domainadminpassword'])
-                ident.identification.joindomain = str(self.params['customizations']['joindomain'])
-            elif 'joinworkgroup' in self.params['customizations']:
-                ident.identification.joinworkgroup = str(self.params['customizations']['joinworkgroup'])
+                ident.identification.domainadmin = str(self.params['customizations'].get('domainadmin'))
+                ident.identification.domainadminpassword = str(self.params['customizations'].get('domainadminpassword'))
+                ident.identification.joindomain = str(self.params['customizations'].get('joindomain'))
+            else:
```

Comment:
I thought about it and I would prefer to not set default value for this param. It could be better to explicitly enable joindomain or joinworkgroup.